pkg/dependencies/factory.go

```diff
@@ -250,6 +257,16 @@ func (f *Factory) WithExecutableBuilder() *Factory {
 		}
 
 		if f.executablesConfig.useDockerContainer {
+
+			if f.registryAuth {
+				username := os.Getenv("REGISTRY_USERNAME")
+				password := os.Getenv("REGISTRY_PASSWORD")
+				err := f.executablesConfig.dockerClient.Login(context.Background(), f.registryMirror, username, password)
+				if err != nil {
+					return err
```

Comment:
you can re-use https://github.com/aws/eks-anywhere/blob/main/pkg/config/registry.go#L8